guides/developer/provider-setups/aws-dev-setup.md

```diff
@@ -0,0 +1,291 @@
+hal config provider aws account edit my-aws-devel-acct --regions $AWS_REGIONS  # regions you want Spinnaker to deploy to
+hal config provider aws enable
+
+# Connect your Docker registries
+hal config provider docker-registry enable
+
+hal config provider docker-registry account add my-dockerhub-devel-acct \
+    --address index.docker.io \
+    --repositories {your dockerhub username}/{your dockerhub repository} \
+    --username {your dockerhub username} \
+    --password \
+    --track-digests true
+
+# Replace the address specified below (123456789012.dkr.ecr.us-west-2.amazonaws.com) with the address to your own ecr registry.
+hal config provider docker-registry account add my-us-west-2-devel-registry \
+ --address 123456789012.dkr.ecr.us-west-2.amazonaws.com \
+ --username AWS \
+ --password-command "aws --region us-west-2 ecr get-authorization-token --output text --query 'authorizationData[].authorizationToken' | base64 -d | sed 's/^AWS://'" \
+ --track-digests true
+```
+
+Optionally connect Amazon ECS to your AWS account
+```
+hal config provider ecs account add ecs-my-aws-devel-acct --aws-account my-aws-devel-acct
+hal config provider ecs enable
+```
+
+Deploy everything
+```
+hal deploy apply
+```
+
+Wait for Clouddriver to start up by checking the logs in ~/dev/spinnaker/logs/clouddriver.log.
+
+# Daily development
+## Test Spinnaker Code Changes
+
+Sync your changes to the development instance with `rsync`:
+```
+rsync --progress -a -e "ssh -i /path/to/my-key-pair.pem" --exclude='*/build/' --exclude='*/.idea/' --exclude='*/out/' --exclude='*/.gradle/' ~/code/spinnaker/ ubuntu@$SPINNAKER_INSTANCE_DNS:/home/ubuntu/dev/spinnaker
+
+# Optional:
+ssh ubuntu@$SPINNAKER_INSTANCE_DNS 'for i in ~/dev/spinnaker/*; do (cd $i && echo $i && git checkout master && git clean -fdx); done'
+```
+
+* If `rsync` isn't an option, you can also push your changes to a remote branch of your fork(s) and manually pull them down to your dev instance before deploying.
+
+[Login to the instance](https://docs.aws.amazon.com/quickstarts/latest/vmlaunch/step-2-connect-to-instance.html), deploy the changes, and check for build or service failures:
+```
+ssh -A -L 9000:localhost:9000 -L 8084:localhost:8084 -L 8087:localhost:8087 ubuntu@$spinnaker_instance -i /path/to/my-key-pair.pem
+
+hal deploy apply
+# or for individual service changes: hal deploy apply --service-names=clouddriver,deck,{other_service_with_changes}
+
+cd ~/dev/spinnaker/logs
+```
+
+Test your changes manually at http://localhost:9000.
+
+## Regularly sync from upstream
+
+Add the following to your laptop's .bashrc file
+```
+sync-from-upstream() {
+    for i in ./*; do
+        (cd $i && echo $i && git checkout master && git pull --rebase upstream master && git push origin upstream/master:master)
+    done
+}
+```
+
+Then regularly run `sync-from-upstream` in your `/spinnaker` directory to keep your local repos and GitHub forks in sync with upstream Spinnaker.
```

Comment:
```suggestion
Regularly run `sync-from-upstream` in your `/spinnaker` directory to keep your local repos and GitHub forks in sync with upstream Spinnaker.
```

Reply:
How often is "regulary"? Like, daily?

Reply:
Added a sentence w/ suggested frequency and brief reasoning